### PR TITLE
[CHORE] update logging and improve cash match

### PR DESCRIPTION
**Pull request by chelsea-EYDS**
[CHORE](https://bcdevex.atlassian.net/browse/CCFPCM-0000)

**Objective:** 

Clean up the current cash match. Match only one deposit date per date per location. 

Improve logging.

New output (Deposit ID can be used to query the matching payments for testing): 

Screenshot 2023-03-13 at 12 59 23 PM

### CODE REVIEW

File: apps/backend/src/deposits/cash-deposit.service.ts [hunk 1]
```diff
@@ -93,11 +93,8 @@ export class CashDepositService {
     const distinctDepositDates = Array.from(
       new Set(dates.map((item) => item.deposit_date))
     );
-    if (distinctDepositDates.length < 4)
-      return {
-        currentDate: date,
-        pastDueDate: null
-      };
+    if (distinctDepositDates.length < 4) return;
```
Comment (chelsea-EYDS) [inline apps/backend/src/deposits/cash-deposit.service.ts:96]: Should this be < 3?

File: apps/backend/src/deposits/cash-deposit.service.ts [hunk 2]
```diff
@@ -113,27 +110,50 @@ export class CashDepositService {
...
+    const { fiscal_close_date, fiscal_start_date, program, location } = event;
 
     const dates = await this.cashDepositRepo.find({
       select: { deposit_date: true },
       where: {
-        pt_location_id,
+        pt_location_id: location.pt_location_id,
         metadata: { program },
-        deposit_date: LessThanOrEqual(date)
+        deposit_date: Raw(
```
Comment (chelsea-EYDS) [inline apps/backend/src/deposits/cash-deposit.service.ts:120]: Set a range, vs less than

File: apps/backend/src/deposits/cash-deposit.service.ts [hunk 3]
```diff
@@ -113,27 +110,50 @@ export class CashDepositService {
...
+   * @param status: MatchStatus
+   * @returns CashDepositEntity
+   */
+  async updateDeposits(
+    deposits: CashDepositEntity[],
+    status: MatchStatus
+  ): Promise<CashDepositEntity[]> {
+    this.appLogger.log(
+      `UPDATED: ${deposits.length} CASH DEPOSITS to ${status.toUpperCase()}`,
+      CashDepositService.name
```
Comment (chelsea-EYDS) [inline apps/backend/src/deposits/cash-deposit.service.ts:147]: Moved out of the cash-reconciliation file

File: apps/backend/src/lambdas/reconcile.ts [hunk 4]
```diff
@@ -22,107 +25,134 @@ export const handler = async (
...
-      fiscalDates.push(new Date(fiscalStart).toDateString());
-      fiscalStart.setDate(fiscalStart.getDate() + 1);
-    }
-    return fiscalDates.map((itm) => itm.toString());
-  };
+  appLogger.log(`=========================================================`);
+  appLogger.log(`Soft Removing Zero Dollar Payments`);
+  appLogger.log(`=========================================================`);
+
+  await paymentService.softRemoveZeroDollarPayments();
```
Comment (chelsea-EYDS) [inline apps/backend/src/lambdas/reconcile.ts:32]: Archive $0.00 payments? (pending team approval)

**Review by eyjwarren — APPROVED** [review]

**Pull request merged**